### DISCUSSION TITLE
Refresh page when double click table

### DISCUSSION THREAD

**Discussion by shamilh2o** [h2oai/wave#1669] [Q&A]
Hey Team,

When I double-click text on the table in order to select it, the table refreshes and then I lose my search filter. 
Is this the normal behaviour? Any suggestion to overcome this and stop refreshing the page when I double click text on the table? 

Thank you.

cc: @mturoci

**Comment by mtanco** [comment]
> If multiple is set to False (default), each row in the table is clickable. When a row is clicked, the form is submitted automatically, and q.args.table_name is set to [row_name], where table_name is the name of the table, and row_name is the name of the row that was clicked on.

@mturoci probably worth adding to doc string that `row is clicked` means either clicking the link column OR double clicking. 

@shamilh2o this is expected behavior - the app is not refreshing per se, it is sending a value back to the `serve` function in the form of `q.args.TABLENAME:['ROW NAME CLICKED']`. It's likely that your current app code creates the table in the serve function with no checks. 

If you want users to pick a single row, double clicking the row is probably okay, yes? And if you want them to choose more than one row then you can just set `multiple=True` which will disable this feature. 

Some repro code for anyone that wants to look:

```python
from h2o_wave import main, app, Q, ui


@app('/')
async def serve(q: Q):
    print(q.args)

    if not q.client.initialized:
        q.page["table_card"] = ui.form_card(
            box="1 1 -1 -1",
            items=[
                ui.table(
                    name="table_test",
                    columns=[ui.table_column(x, x) for x in ["test1", "test2"]],
                    rows=[ui.table_row(str(i), ["hey", "ho"]) for i in range(10)],
                    height="600px",
                )
            ]
        )
        q.client.initialized = True

    await q.page.save()
```

**Reply by shamilh2o** [reply]
Thanks for the clear answer @mtanco. 
What if the user needs to copy a content in a table cell by simply after double clicking and highlighting it? That's what my actual use case is. Can we do it without sending value back to the serve function? (Other than enabling multiple = True) 

https://user-images.githubusercontent.com/101506482/195814195-c53d40f1-8106-4e90-96d5-a131adba439c.mov

**Reply by mturoci** [reply]
Good point @mtanco, updated in https://github.com/h2oai/wave/commit/6b2c2da1faf57f47e566a8f6e7fdfdadf22ded24. Thanks!

**Reply by Far0n** [reply]
sounds related to: https://github.com/h2oai/wave/issues/216

**Reply by shamilh2o** [reply]
Thanks @Far0n for jumping over here. What I actually need to do is not to select a single row. Just want to copy a content in a row cell (like the value 8361511 in video above). In that case if we double click on a text to highlight it before pressing "command + c" to copy, it cannot do since once we double click, table refreshes by sending value back to the serve function. 

Can I set not to send table value to serve function when I double click on a row?

**Reply by mturoci** [reply]
This is fairly simple to implement. Can you please file a feature request @shamilh2o? Thanks!

**Reply by shamilh2o** [reply]
Sure @mturoci. Thanks!
